### PR TITLE
Tdd tut #155730453

### PR DESCRIPTION
**Pull request by MidWeekUser**
## Description
This PR is for learning how to use tests when creating APIs.
## Tasks Completed:
- Created folder structure.
- Created test files in 'tests' folder.
- Ran failing tests to verify of the run.
- Created simple API apps in 'api' folder.
- Ran tests to verify the code.
- Debugged errors to run successful tests.
## Manual Testing (using Bash)
- Download all files to a directory
- Install requirements in 'requirements_flusk_restfulpus.txt' file
- Navigate to root folder
- Use pytest to run the test files
## Pivotal Tracker
Click [here](https://www.pivotaltracker.com/story/show/155730453) to see the story relevant to this PR.

### CODE REVIEW

File: app/api/users.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from flask import Flask
+from flask_restplus import Resource, Api
+
+app = Flask(__name__)
+api = Api(app)
+# @api.route('/hello')
```
Comment (georgreen) [inline app/api/users.py]: Remove comments from your work before you push.

Comment (MidWeekUser) [reply app/api/users.py]: Will do...

File: app/api/user_management.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+UserObj = UserManagement()
+
+class RegisterUsers(Resource):
+    
+    
+    
+    def post(self):
+        
+        user_data = request.json
```
Comment (georgreen) [inline app/api/user_management.py]: You are not validating the data received from the client. Please try to incorporate this.

File: app/api/user_management.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        email = user_data['email']
+        firstname = user_data['firstname']
+        secondname = user_data['secondname']
+        username = user_data['username']
```
Comment (georgreen) [inline app/api/user_management.py]: This code is unsafe and prone to bugs, accessing dictionary using the square bracket can throw exceptions when the key accessed is missing. Correct this by checking the key exists in the dictionary.

File: app/api/user_management.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        password = user_data['password']
+        confirmpassword = user_data['confirmpassword']
+
+        status = UserObj.registernewuser(email, firstname, secondname, username, password, confirmpassword)
+        
+        return status
+    
+    def get(self):
```
Comment (georgreen) [inline app/api/user_management.py]: What is a get Verb for in a register resource?

File: app/api/user_management.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        return {'Alert': 'Get Successful'}
+
+class RegisterNewBusiness(Resource):
+    def post(self):
+        return {'Alert': 'Post Successful'}
+    
+
+# class Login(Resource):
+#     from app.models import UserManagement
```
Comment (georgreen) [inline app/api/user_management.py]: You should remove unused code from your work before you push.

File: app/tests/test_user_management.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import json
+
+from WeConnect.app import app_factory
+
+app = app_factory('TestingConfig')
+
+class user_scenarios(unittest.TestCase):
+    def setUp(self):
+        self.postman = app.test_client()
+        self.header = {'Content-type': 'application/json'}
```
Comment (georgreen) [inline app/tests/test_user_management.py]: This is incorrect, setting the content-type should not be done in the header.

File: run.py [hunk 7]
```diff
@@ -1,6 +1,8 @@
 # run.py
 
-from app import app
+from app import app_factory
+
+app = app_factory('DevelopmentConfig')
```
Comment (georgreen) [inline run.py]: You should not hard code environments, this should be picked from the shell environment. This makes your app more flexible.

**Review by georgreen — COMMENTED** [review]
You should not push your virtual env to git, please rectify this.

**Pull request merged**